src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageDetailsTabViewModel.cs

```diff
@@ -0,0 +1,106 @@
+
+        public ReadmePreviewViewModel ReadmePreviewViewModel { get; private set; }
+
+        public DetailControlModel DetailControlModel { get; private set; }
+
+        public ObservableCollection<RenderedViewModelBase> Tabs { get; private set; }
+
+        private RenderedViewModelBase _selectedTab;
+        public RenderedViewModelBase SelectedTab
+        {
+            get => _selectedTab;
+            set
+            {
+                SetAndRaisePropertyChanged(ref _selectedTab, value);
+            }
+        }
+
+        public PackageDetailsTabViewModel()
+        {
+            _readmeTabEnabled = true;
+            Tabs = new ObservableCollection<RenderedViewModelBase>();
+        }
+
+        public PackageMetadataTab GetSelectedTab()
+        {
+            return SelectedTab is DetailControlModel ? PackageMetadataTab.PackageDetails : PackageMetadataTab.Readme;
+        }
+
+        public async Task InitializeAsync(DetailControlModel detailControlModel, INuGetPackageFileService nugetPackageFileService, ItemFilter currentFilter, PackageMetadataTab initialSelectedTab)
+        {
+            var nuGetFeatureFlagService = await ServiceLocator.GetComponentModelServiceAsync<INuGetFeatureFlagService>();
+            _readmeTabEnabled = await nuGetFeatureFlagService.IsFeatureEnabledAsync(NuGetFeatureFlagConstants.RenderReadmeInPMUI);
+
+            ReadmePreviewViewModel = new ReadmePreviewViewModel(nugetPackageFileService, currentFilter);
+            ReadmePreviewViewModel.IsVisible = _readmeTabEnabled;
+            DetailControlModel = detailControlModel;
+
+            Tabs.Add(ReadmePreviewViewModel);
+            Tabs.Add(DetailControlModel);
+
+            SelectedTab = _readmeTabEnabled && initialSelectedTab == PackageMetadataTab.Readme
+                ? ReadmePreviewViewModel
+                : DetailControlModel;
+
+            DetailControlModel.PropertyChanged += DetailControlModel_PropertyChanged;
+            ReadmePreviewViewModel.PropertyChanged += ReadMePreviewViewModel_PropertyChanged;
+        }
+
+        public async Task SetCurrentFilterAsync(ItemFilter filter)
+        {
+            if (_readmeTabEnabled)
+            {
+                await ReadmePreviewViewModel.SetCurrentFilterAsync(filter);
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
+            DetailControlModel.PropertyChanged -= DetailControlModel_PropertyChanged;
+        }
+
+        private void ReadMePreviewViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (ReadmePreviewViewModel is not null
+                && e.PropertyName == nameof(ReadmePreviewViewModel.IsVisible))
+            {
+                if (!ReadmePreviewViewModel.IsVisible)
+                {
+                    SelectedTab = DetailControlModel;
+                }
+            }
+        }
```

Comment:
This could be rewritten to more generally handle when a tab goes invisible and make the next one in line, or first one from the start of the collection (depending on how we want it to behave), that has an IsVisible = true, the selected tab. That way we avoid some code that specifically deals with if ReadmePreviewViewModel is invisible, then select the DetailControlModel, and instead, we say if the selected tab goes invisible, select the next one. We will also need to hook this same event handler up to all the Tabs in the collection.

Reply:
Okay updated it! I check to see if the currently selected tab is visible and if it isn't then switch to the first avaiable visible tab.